### ISSUE TITLE
Confusing action limitation on intentions

### ISSUE THREAD

**Issue by 3nprob** [hashicorp/consul#11012]
**Old UI or New UI**
New UI

**Describe the problem you're having**
Intention created directly via Terraform provider. Actions -> "This intention is being managed through an Intention Custom Resource in your Kubernetes cluster. It is view only in the UI."

There has never been a Kubernetes cluster or configuration in this environment.

![Screenshot_2021-09-10 Intentions - Consul](https://user-images.githubusercontent.com/74199244/132902917-fcf4c971-2e9f-499f-a1f4-a58b909c593a.png)

**Describe the solution you'd like**
See available actions, e.g. "delete"

**Comment by johncowen** [comment]
Hi @3nprob 

Thanks for the report! Would it be possible for you to post the response from:

`/v1/connect/intentions?dc=datacenter-name`

When you see this happening in the UI? You can get this information from opening the Web Inspector in your browser and clicking the Network panel. In there you should be able to copy out the response from that API request.

Thanks,

**Comment by reskin89** [comment]
Also seeing this on consul 1.10.2.

The endpoint returns the intention and it looks normal:
```json
  {
    "SourceNS": "default",
    "SourceName": "REDACTED-SOURCE",
    "DestinationNS": "default",
    "DestinationName": "REDACTED-DESTINATION",
    "SourceType": "consul",
    "Action": "allow",
    "Precedence": 9,
    "CreateIndex": 4486909,
    "ModifyIndex": 6421921
  }
  ```

**Comment by johncowen** [comment]
Hi @reskin89 👋 

Is that the response from single intention endpoint or the intentions listing endpoint?

**Comment by reskin89** [comment]
It's the response from the intentions endpoint, I pasted one of the returns I'm seeing this message on. They all look similar aside from the time stamps and source/dest names.

It's all just an array of these objects no other info

**Comment by johncowen** [comment]
Ok cool, thanks @reskin89 . Can I get some more information from you, I'm having a tough time reproducing this.

What browser/version are you using?

**Comment by reskin89** [comment]
Thanks John!

We're seeing it in Firefox 92 and Chrome Version 93.0.4577.82

Oddly we're also noticing tokens with the write permissions on intentions aren't able to create intentions from the topology screen with the red X like they should.  It believes the token they have is read only and we've confirmed it is not.

I'm not sure if they're related

**Comment by johncowen** [comment]
Hmm yeah, there's an ongoing issue at the moment related to the API we use with permissions. It shouldn't be related, but is confusing this nonetheless. Lemme look some more.

Would it be possible for you to paste me in the entire intentions listing endpoint HTTP response (redacted of course) when you are seeing this "Managed By CRD" message and I can try and repro some more.

**Comment by GordonMcKinney** [comment]
@johncowen I work with @reskin89

Clicking any "..." on the intentions screen produces the same CRD error.

The policy for this token is:

```
service_prefix "A-PREFIX-" {
  policy = "write"
  intentions = "write"
}
node_prefix "" {
  policy = "read"
}
service_prefix "" {
  policy = "read"
  intentions = "write"
}
```

Here's the intentions endpoint response:
```
[
    {
        "SourceNS": "default",
        "SourceName": "<redacted>,
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 9,
        "CreateIndex": 4486909,
        "ModifyIndex": 6421921
    },
    {
        "SourceNS": "default",
        "SourceName": "<redacted>,
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 9,
        "CreateIndex": 6146047,
        "ModifyIndex": 6420003
    },
    {
        "SourceNS": "default",
        "SourceName": "*",
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 8,
        "CreateIndex": 339590,
        "ModifyIndex": 6307988
    },
    {
        "SourceNS": "default",
        "SourceName": "*",
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 8,
        "CreateIndex": 1854566,
        "ModifyIndex": 6307990
    },
    {
        "SourceNS": "default",
        "SourceName": "*",
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 8,
        "CreateIndex": 1391340,
        "ModifyIndex": 6307991
    },
    {
        "SourceNS": "default",
        "SourceName": "*",
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 8,
        "CreateIndex": 5496565,
        "ModifyIndex": 6307992
    },
    {
        "SourceNS": "default",
        "SourceName": "*",
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 8,
        "CreateIndex": 5350751,
        "ModifyIndex": 6307993
    },
    {
        "SourceNS": "default",
        "SourceName": "*",
        "DestinationNS": "default",
        "DestinationName": "<redacted>,
        "SourceType": "consul",
        "Action": "allow",
        "Precedence": 8,
        "CreateIndex": 1390828,
        "ModifyIndex": 6307994
    }
]
```

**Comment by johncowen** [comment]
Perfect thanks @GordonMcKinney !

I loaded this exact response (well almost exact, I had to make the redacted things unique as you can't have two intentions exactly the same) into the UI at our v1.10.2 tag and this is what I see:

<img width="1378" alt="Screenshot 2021-09-24 at 18 25 14" src="https://user-images.githubusercontent.com/554604/134716327-e8ccc9af-9484-48a8-8e43-e6061c635779.png">

So this is curious. Trying to think of what differences in the repro there could be here 🤔

**Comment by johncowen** [comment]
P.S. are you seeing the CRD message as soon as you initially visit the page from a refresh (and then click the [...])? Another thing maybe, are there any red error messages in the Web Inspector console when you see the CRD message?

**Comment by johncowen** [comment]
Ah ah ah, wait one second... 😂 

🤞 

Brb

**Comment by johncowen** [comment]
@GordonMcKinney would you be able to pass me your response from the `/v1/internal/acl/authorize?dc=<your-dc>` endpoint?

**Comment by johncowen** [comment]
P.P.S. I think this PR (not in 1.10.2 to fix an issue introduced in 1.10.2) would go at least some way to solve this issue

https://github.com/hashicorp/consul/pull/11061

I'll look further into this after the weekend, but extra bug thanks for keeping the feedback coming here. Its been a big help not just for this issue but for others that are appearing 🙏 

cc @jkirschner-hashicorp

**Comment by GordonMcKinney** [comment]
https://.../v1/internal/acl/authorize?dc=us-east-1
```
[
    {
        "Resource": "operator",
        "Access": "write",
        "Allow": false
    },
    {
        "Resource": "operator",
        "Access": "read",
        "Allow": false
    },
    {
        "Resource": "service",
        "Access": "read",
        "Allow": true
    },
    {
        "Resource": "node",
        "Access": "read",
        "Allow": true
    },
    {
        "Resource": "session",
        "Access": "read",
        "Allow": false
    },
    {
        "Resource": "session",
        "Access": "write",
        "Allow": false
    },
    {
        "Resource": "key",
        "Access": "read",
        "Allow": false
    },
    {
        "Resource": "key",
        "Access": "write",
        "Allow": false
    },
    {
        "Resource": "intention",
        "Access": "read",
        "Allow": true
    },
    {
        "Resource": "intention",
        "Access": "write",
        "Allow": false
    },
    {
        "Resource": "acl",
        "Access": "read",
        "Allow": false
    },
    {
        "Resource": "acl",
        "Access": "write",
        "Allow": false
    }
]
```




https://.../v1/internal/acl/authorize?dc=us-east-1
```
[
  {
    "Resource": "service",
    "Segment": "<redacted>",
    "Access": "read",
    "Allow": true
  },
  {
    "Resource": "service",
    "Segment": "<redacted>",
    "Access": "write",
    "Allow": true
  }
]
```

**Comment by GordonMcKinney** [comment]
Here's a diff of the user vs master token authorize response. I've highlighted the intention ability.

<img width="1096" alt="Screen Shot 2021-09-24 at 2 30 47 PM" src="https://user-images.githubusercontent.com/4955046/134730174-9beb801b-5363-4e62-9b48-582ec8d037c4.png">

**Comment by johncowen** [comment]
Hi all, PR up for this ^.

Thanks again for the help here! 👏 👏 👏 🙇 🙇 🙇

**Comment by GordonMcKinney** [comment]
Thank you for getting this resolved so quickly!

**Comment by GordonMcKinney** [comment]
Confirmed fixed in 1.10.3
